tests/checkpointing/test_model_checkpoint.py

```diff
@@ -247,14 +247,7 @@ def configure_optimizers(self):
     ckpt_files = list(Path(tmpdir).glob('*.ckpt'))
     lr_scheduler_debug = trainer.dev_debugger.saved_lr_scheduler_updates
 
-    # on_train_end ckpt callback is called which creates an additional ckpt in case no ckpt is created at the
-    # end of epoch, thus if val_check_interval doesn't align with the training steps we create an additional ckpt
```

Comment:
does this not apply anymore because we have a check `_should_skip_saving_checkpoint` in the `ModelCheckpoint. on_train_end` ?

Reply:
No, as `check_checkpoint_callback` called `on_validation_end` which also has `should_skip_saving_checkpoint`.

The reason is that the global step has increased when we check to save `on_train_end` when not aligned, and since the `on_validation_end` hook is used which does not early exit if `save_last=False`, it saves another checkpoint.

So we were forcing an extra "last" checkpoint which was not technically a `save_last` checkpoint